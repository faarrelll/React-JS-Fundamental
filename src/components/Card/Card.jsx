export default function Card({
  title,
  description,
  tags = [],
  status = "draft",
}) {
  return (
    <div className="border rounded-lg p-4 shadow-sm">
      <div className="flex justify-between items-center mb-2">
        <h2 className="text-xl font-bold">{title}</h2>
        <span
          className={`px-2 py-1 rounded-full text-sm ${
            status === "published"
              ? "bg-green-100 text-green-800"
              : status === "review"
              ? "bg-gray-100 text-gray-800"
              : "bg-yellow-100 text-yellow-800"
          }`}
        >
          {status}
        </span>
      </div>
      <p className="text-gray-600 mb-4">{description}</p>
      <div className="flex gap-2">
        {tags.map((tag) => (
          <span
            key={tag}
            className="bg-blue-100 text-blue-800 px-2 py-1 rounded text-sm"
          >
            {tag}
          </span>
        ))}
      </div>
    </div>
  );
}
