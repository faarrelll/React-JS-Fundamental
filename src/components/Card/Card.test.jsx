import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Card from "./Card";

describe("Card Component Snapshots", () => {
  it("renders default draft card correctly", () => {
    const { container } = render(
      <Card
        title="Getting Started with Vitest"
        description="Learn how to test React components"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders published card with tags correctly", () => {
    const { container } = render(
      <Card
        title="Advanced Testing Tips"
        description="Master React testing strategies"
        status="published"
        tags={["react", "testing", "vitest"]}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders card with review status correctly", () => {
    const { container } = render(
      <Card
        title="Work in Progress"
        description="This post is under review"
        status="review"
        tags={["draft"]}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
