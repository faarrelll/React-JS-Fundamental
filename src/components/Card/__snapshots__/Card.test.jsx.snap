// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component Snapshots > renders card with review status correctly 1`] = `
<div>
  <div
    class="border rounded-lg p-4 shadow-sm"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <h2
        class="text-xl font-bold"
      >
        Work in Progress
      </h2>
      <span
        class="px-2 py-1 rounded-full text-sm bg-gray-100 text-gray-800"
      >
        review
      </span>
    </div>
    <p
      class="text-gray-600 mb-4"
    >
      This post is under review
    </p>
    <div
      class="flex gap-2"
    >
      <span
        class="bg-blue-100 text-blue-800 px-2 py-1 rounded text-sm"
      >
        draft
      </span>
    </div>
  </div>
</div>
`;

exports[`Card Component Snapshots > renders default draft card correctly 1`] = `
<div>
  <div
    class="border rounded-lg p-4 shadow-sm"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <h2
        class="text-xl font-bold"
      >
        Getting Started with Vitest
      </h2>
      <span
        class="px-2 py-1 rounded-full text-sm bg-yellow-100 text-yellow-800"
      >
        draft
      </span>
    </div>
    <p
      class="text-gray-600 mb-4"
    >
      Learn how to test React components
    </p>
    <div
      class="flex gap-2"
    />
  </div>
</div>
`;

exports[`Card Component Snapshots > renders published card with tags correctly 1`] = `
<div>
  <div
    class="border rounded-lg p-4 shadow-sm"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <h2
        class="text-xl font-bold"
      >
        Advanced Testing Tips
      </h2>
      <span
        class="px-2 py-1 rounded-full text-sm bg-green-100 text-green-800"
      >
        published
      </span>
    </div>
    <p
      class="text-gray-600 mb-4"
    >
      Master React testing strategies
    </p>
    <div
      class="flex gap-2"
    >
      <span
        class="bg-blue-100 text-blue-800 px-2 py-1 rounded text-sm"
      >
        react
      </span>
      <span
        class="bg-blue-100 text-blue-800 px-2 py-1 rounded text-sm"
      >
        testing
      </span>
      <span
        class="bg-blue-100 text-blue-800 px-2 py-1 rounded text-sm"
      >
        vitest
      </span>
    </div>
  </div>
</div>
`;
